### PR TITLE
WINC-1117: Pick up openshift/kubernetes 1.28 rebase updates

### PR DESCRIPTION
**Pull request by aravindhp**
- [submodule][kubelet] Update to e3ba6d9fbe7
- [build] Update kubelet version to v1.28.2+e3ba6d9
- Set v1.28 as minimum k8s version
- [vendor] Use 1.28.2 k8s and OpenShift libraries
- Sync with ctrl runtime changes
- [test] Sync Azure machine-api provider changes

### CODE REVIEW

**Comment by aravindhp** [comment]
/test aws-e2e-upgrade

Install failure

**Comment by mansikulkarni96** [comment]
Thanks for working on this @aravindhp 
/lgtm

**Review by jrvaldes — COMMENTED** [review]
@aravindhp  thanks for working on this. LGTM

Is [4.15.0-0.ci](https://amd64.ocp.releases.ci.openshift.org/#4.15.0-0.ci) already in 1.28 or should we hold?

File: cmd/operator/main.go [hunk 1]
```diff
@@ -148,9 +150,13 @@ func main() {
...
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:             scheme,
-		MetricsBindAddress: fmt.Sprintf("%s:%d", metrics.Host, metrics.Port),
-		Port:               9443,
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress: fmt.Sprintf("%s:%d", metrics.Host, metrics.Port),
+		},
+		WebhookServer: webhook.NewServer(webhook.Options{
+			Port: 9443,
```
Comment (sebsoto) [inline cmd/operator/main.go:158]: Not sure if you saw, this is the default value https://github.com/kubernetes-sigs/controller-runtime/blob/c93e2abcb28eb71fccad7dc565f0547cc07e5566/pkg/webhook/server.go#L78

**Review by sebsoto — APPROVED** [review]

File: cmd/operator/main.go (hunk 1 above)
Comment (aravindhp) [reply cmd/operator/main.go:158]: I missed that. I will remove this line then.

**Comment by aravindhp** [comment]
Nutanix job [failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1845/pull-ci-openshift-windows-machine-config-operator-master-nutanix-e2e-operator/1708207557006331904) with the following error:
```
2023-09-30T22:39:44Z	ERROR	Reconciler error	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap", "ConfigMap": {"name":"windows-services-10.15.0-d65a749-dirty","namespace":"openshift-windows-machine-config-operator"}, "namespace": "openshift-windows-machine-config-operator", "name": "windows-services-10.15.0-d65a749-dirty", "reconcileID": "0cbfd3bf-03e6-411e-ac13-603659639852", "error": "unable to create signer from private key secret: Secret \"cloud-private-key\" not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/build/windows-machine-config-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
```
Not sure how that is possible given we create the secret as part of launching the operator using the e2e script

**Comment by aravindhp** [comment]
/override ci/prow/vsphere-proxy-e2e-operator

Tests passed but job failed in deprovision:
```
level=error msg=Folder should be empty but contains 1 objects: VirtualMachine:vm-833984. The installer will retry removing "virtualmachine" objects, but any other type will need to be removed manually before the deprovision can proceed Folder=ci-op-b5grf1lg-7c60a-9blpr
```
Seen this error twice now. If it continues, we need to look into this.

**Pull request merged**